llvm/lib/IR/Type.cpp

```diff
@@ -845,6 +852,19 @@ static TargetTypeInfo getTargetTypeInfo(const TargetExtType *Ty) {
     return TargetTypeInfo(ScalableVectorType::get(Type::getInt1Ty(C), 16),
                           TargetExtType::HasZeroInit);
 
+  // RISC-V vector tuple type. The layout is represented as the type that needs
+  // the same number of vector registers(VREGS) as this tuple type, represented
+  // as <vscale x (VREGS * 8) x i8>.
+  if (Name == "riscv_vec_tuple") {
```

Comment:
Is there a mechanism to avoid strcmp? We can generate category ID for target extension types.

Reply:
And, I think we should rename `riscv_vec_tuple` to `riscv.vector.tuple`. Though there is no naming convention for target exension types now, I think we should follow the intrinsic naming rule.

Reply:
> Is there a mechanism to avoid strcmp? We can generate category ID for target extension types.

Yeah, I think it's a good idea, let me think about it, thanks!

Reply:
> And, I think we should rename `riscv_vec_tuple` to `riscv.vector.tuple`. Though there is no naming convention for target exension types now, I think we should follow the intrinsic naming rule.

You are right, it's more consistent and readable.

Reply:
Issue created: https://github.com/llvm/llvm-project/issues/99440